components/profile_popover/index.ts

```diff
@@ -24,15 +24,17 @@ import {areTimezonesEnabledAndSupported} from 'selectors/general';
 import {getRhsState, getSelectedPost} from 'selectors/rhs';
 
 import {makeGetCustomStatus, isCustomStatusEnabled} from 'selectors/views/custom_status';
+import {ActionFunc, GenericAction} from 'mattermost-redux/types/actions';
+import {GlobalState} from '../../types/store';
 
-import ProfilePopover from './profile_popover.jsx';
+import ProfilePopover from './profile_popover';
 
-function getDefaultChannelId(state) {
+function getDefaultChannelId(state: GlobalState) {
     const selectedPost = getSelectedPost(state);
     return selectedPost.exists ? selectedPost.channel_id : getCurrentChannelId(state);
 }
 
-function mapStateToProps(state, {userId, channelId = getDefaultChannelId(state)}) {
+function mapStateToProps(state: GlobalState, {userId, channelId = getDefaultChannelId(state)}: {userId: string; channelId?: string}) {
```

Comment:
Nit: usually we define an actual type called `OwnProps` for the second parameter.
Non-blocking though.

Reply:
Done